### ISSUE TITLE
All replies should be processed as they are available

### ISSUE THREAD

**Issue by artiommorozov** [JoshGlazebrook/socks#58]
On long networks both SOCKS5 bind replies sometimes arrive as a single packet. That causes single invoke of SocksClient::onDataReceivedHandler and only first reply gets processed. The second reply hangs in receiveBuffer unless remote peer sends something or connection times out.

Possible fix would be replacing 
```
  private processData() {
    // If we have enough data to process the next step in the SOCKS handshake, proceed.
    if (this.receiveBuffer.length >= this.nextRequiredPacketBufferSize) {
```

with 
```
  private processData() {
    // If we have enough data to process the next step in the SOCKS handshake, proceed.
    while (this.receiveBuffer.length >= this.nextRequiredPacketBufferSize && this.state !== SocksClientState.Established) {
```

**Comment by JoshGlazebrook** [comment]
Nice catch, I'll try to get a new version released later today!

**Comment by JoshGlazebrook** [comment]
Fixed in 2.4.2
